collector/redfish_collector.go

```diff
@@ -99,6 +99,8 @@ func (r *RedfishCollector) Collect(ch chan<- prometheus.Metric) {
 
 	ch <- r.redfishUp
 	ch <- prometheus.MustNewConstMetric(totalScrapeDurationDesc, prometheus.GaugeValue, time.Since(scrapeTime).Seconds())
+	defer r.redfishClient.Logout()
```

Comment:
Hmm, the logout on line 84 should handle the session close. I hadn't seen that.

We can't do it on line 102 because `r.redfishClient` could be `nil`.

It does appear that we hit the else branch in practise. I see `redfish_up = 0` in historical data, and around those times we see dropouts in other data. We need to get to bottom of why that happens.